projects/fal/src/fal/api.py

```diff
@@ -791,11 +794,26 @@ class BaseServable:
     def collect_routes(self) -> dict[RouteSignature, Callable[..., Any]]:
         raise NotImplementedError
 
-    def _add_extra_middlewares(self, app: FastAPI):
+    def _add_extra_middlewares(self, app: FastAPI) -> None:
         """
         For subclasses to add extra middlewares to the app.
         """
-        pass
+
+        # Common middlewares
+        @app.middleware("http")
+        async def fal_logger_middleware(request, call_next):
+            try:
+                fal_headers = {
+                    header: request.headers[header]
+                    for header in request.headers
+                    if header.startswith("X-Fal-Log-")
+                }
+                # TODO: somehow add the header to the logger context of the served function or app
+            except Exception:
+                logger.exception(
+                    "Failed to get fal headers for %s",
+                    self.__class__.__name__,
+                )
```

Comment:
i think we can't serialize this (`logger`) to the other side as is. we need to do some like the example in the provide hints where we import the logger at the runtime and manually do something.